bin/oneoff/wipe_data/opt_out_petition_emails_under_16

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+# This script sets EmailPreference.opt_in to false for any person who submitted a Pegasus Petition form and was less
+# than 16 years old when they submitted the form because GDPR requires someone to be at least 16 to consent to opt in.
+# Before mid-May 2018, the Petition form anonymized email addresses submitted by people younger than 13.  Starting in
+# mid-May 2018 the Petition form anonymizes email addresses submitted by anyone under 16.  This script does not
+# anonymize or delete email addresses submitted by Petitioners who were between the ages of 13 and 16 because
+# our Contact Rollups process does not currently have any functionality to delete emails from Pardot that were deleted
+# in our database, but it does comply with GDPR by opting out those email addresses.
+
+require File.expand_path('../../../../pegasus/src/env', __FILE__)
+require src_dir 'database'
+load pegasus_dir('helpers.rb')
+require 'cdo/only_one'
+
+def main
+  CDO.log.info "Starting to Opt Out Petition Emails Under 16"
+  # This query loads about 200_000 (200K) rows into memory.
+  DB[:forms].
+    select(:email).
+    where(kind: 'Petition').
+    # Skip Petitions from people under 13, because they were already anonymized.
+    where(Sequel.~(email: 'anonymous@code.org')).
+    where(Sequel.lit("JSON_EXTRACT(data, '$.age_i') < 16")).
+    each do |petition|
+      EmailPreferenceHelper.upsert!(
+        email: petition[:email],
+        opt_in: false,
+        source: EmailPreferenceHelper::AUTOMATED_OPT_OUT_UNDER_16,
+        ip_address: '127.0.0.1',
```

Comment:
Can we use the IP address recorded when the form was saved?

Reply:
We're the ones making the decision to opt them out, so it seems more accurate to log "our" IP address?

Reply:
Oh, good point.  Should we make that a constant indicating that case, and maybe even put it into EmailPreference?

Reply:
Yes.  Done!